### PR TITLE
Jakarta jms

### LINKED ISSUE #3192: JMS message beans doesn't contain elastic apm logging
We upgraded our Spring Boot and Openliberty services to Java 17 and Spring Boot 3. What we've noticed is that the message beans for openliberty and spring boot (JMS) are not instrumenting the elasticsearch properties into the the logging. The rest-api's are logging correctly the elastic properties. We've also tried to print the trace id in our application by ourself with "ElasticApm.currentTransaction().getTraceId()". This is giving us an empty value. 

So are we missing something in the configuration, because of the upgrade? We need the transaction and trace id in the logging. 

The version we use is 1.39.0.

Our properties:
`elastic.apm:
  environment: dev
  service-name: apm-test-logging
  server-url: https://test:8200
  verify-server-cert: false
  enable-log-correlation: true
  log-format-sout: json
  application-packages: nl.test
  central_config: false`

### PR DESCRIPTION
**Pull request by jackshirazi**
Steps:

1. move all javax.jms classes from *.jms to to new *.jms.javax sub-package
2. add jakarta.jms as a dependency to the jms plugin, and refactor the dependencies lower than the top level plugin module
3. copy the *.jms.javax package over to *.jms.jakarta, and switch imports to jakarta.jms (from javax.jms)
4. refactor the two sets of sub-packages to extract out commonalities and put the common base classes in *.jms
5. similarly move the javax tests to the equivalent sub-package
6. copy the javax tests to a new module apm-jms-jakarta-test andimpotr jakarta.jms instead of javax.jms. The new module is primarily because the artemis connection factory is the identical same fully-qualified classname for the two cases (javax.jms & jakarta.jms), but they point at different dependencies and so the pom needs to be different for the two test packages
7. still to do are spring jakarta.jms tests, but let's leave that for another PR

Fixes #3192

### CODE REVIEW

**Comment by SylvainJuge** [comment]
@jackshirazi I've pushed a few changes in ecd1a01b8e96e781edd1c28bc9b86521cb7c4afe after testing with the provided example in https://github.com/elastic/apm-agent-java/issues/3192#issuecomment-1610168171

**Review by SylvainJuge — COMMENTED** [review]
I'm ok with the duplicated version as a first step, but I'm a bit afraid that it would quickly become hard to maintain over time, so I think an extra refactoring step is necessary here.

For example, we should probably move more code to the helper classes, and that would also apply to test code and require to create "test helper" classes or adapters like we have for the servlet API.

File: apm-agent-plugins/apm-jms-plugin/apm-jms-plugin-base/pom.xml [hunk 1]
```diff
@@ -13,6 +13,41 @@
...
+            <artifactId>javax.jms-api</artifactId>
+            <version>2.0</version>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>jakarta.jms</groupId>
+            <artifactId>jakarta.jms-api</artifactId>
+            <version>3.0.0</version>
+            <scope>provided</scope>
+        </dependency>
```
[the remark this answers is not in the document]
Comment (jackshirazi) [reply apm-agent-plugins/apm-jms-plugin/apm-jms-plugin-base/pom.xml:32]: Yes, good suggestion, will do

File: apm-agent-plugins/apm-jms-plugin/apm-jms-plugin-base/src/main/java/co/elastic/apm/agent/jms/BaseJmsInstrumentation.java [hunk 2]
```diff
@@ -19,46 +19,91 @@
...
+        }
     }
 
+    public abstract String rootClassNameThatClassloaderCanLoad();
+
     protected static class BaseAdvice {
 
-        protected static final JmsInstrumentationHelper helper;
-        protected static final MessagingConfiguration messagingConfiguration;
+//        protected static final MessagingConfiguration messagingConfiguration;
```
[the remark this answers is not in the document]
Comment (jackshirazi) [reply apm-agent-plugins/apm-jms-plugin/apm-jms-plugin-base/src/main/java/co/elastic/apm/agent/jms/BaseJmsInstrumentation.java:101]: yes, I forgot to

**Review by SylvainJuge — APPROVED** [review]

**Pull request merged**